crates/rooch-rpc-server/src/service/rpc_service.rs

```diff
@@ -113,7 +113,7 @@ impl RpcService {
         access_path: AccessPath,
         cursor: Option<Vec<u8>>,
         limit: usize,
-    ) -> Result<Vec<Option<(Vec<u8>, State)>>> {
+    ) -> Result<Vec<(Vec<u8>, State)>> {
```

Comment:
Is it that the format vec returned by all list methods no longer contains option, or that the vct returned by all rpc methods no longer contains option?
That is, if the rpc result is a vec structure, the format is vec<T> or vec<option <T>>, I feel that it needs to be unified. The underlying database returns vec<option<T>> by default.

Reply:
I have change the `smt::list` to return `Vec<(K,V)>`

Reply:
Not only the states API, but also other APIs, such as the transaction API, the return value is Vec<Option<TransactionWithInfoView>>

if the rpc result is a vec structure, the format is vec\<T\> or vec<option\<T\>>, need unified?

Reply:
It needs to be handled case by case.

Such as `GetTxExecutionInfosByHashMessage`, if the ExecutionInfo of `tx_hash` does not exist, it should be Option. We need to ensure the TxHash length equals the result Vec length.

Reply:
`tx_hash` is valid but ExecutionInfo does not exist. This situation should not exist, right? The current processing will directly throw an error `Invalid tx hash or tx order`

Reply:
Raising an error should also be a solution. if we raise error, the result should be Vec<T>, not Vec<Option<T>>

Reply:
I record an issue #966